tests/e2e_test.go

```diff
@@ -1127,3 +1166,111 @@ func TestEcdsaHexToPEMMigration(t *testing.T) {
 	}
 	verifyTuf(t, td, meta["root.json"])
 }
+
+// Tests snapshot key rotation.
+func TestSnapshotKeyRotate(t *testing.T) {
+	ctx := context.Background()
+	td := t.TempDir()
+
+	rootCA, rootSigner, err := CreateRootCA()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testTarget := filepath.Join(td, "foo.txt")
+	targetsConfig := map[string]json.RawMessage{testTarget: nil}
+
+	if err := os.WriteFile(testTarget, []byte("abc"), 0600); err != nil {
+		t.Fatal(err)
+	}
+
+	rootSerial, err := CreateTestHsmSigner(td, rootCA, rootSigner)
+	if err != nil {
+		t.Fatal(err)
+	}
+	snapshotKey1 := createTestSigner(t)
+	timestampKey := createTestSigner(t)
```

Comment:
Seen very similar inot code a lot now, would it make sense to create a method for prepare such a test fixture? Where eg a list of "names" are provided and a test signer is created for that, list of target files etc.

Reply:
thanks! yes! updated to use a repo test stack: it can be improved a little, but it's a start